AWSCore/Utility/AWSSynchronizedMutableDictionary.m

```diff
@@ -19,52 +19,73 @@ @interface AWSSynchronizedMutableDictionary()
 
 @property (nonatomic, strong) NSMutableDictionary *dictionary;
 @property (nonatomic, strong) dispatch_queue_t dispatchQueue;
+@property (nonatomic, strong) NSUUID *syncKey;
+@property (readwrite, nonatomic, strong) NSUUID *instanceKey;
 
 @end
 
 @implementation AWSSynchronizedMutableDictionary
 
 - (instancetype)init {
-    if (self = [super init]) {
-        _dictionary = [NSMutableDictionary new];
-        _dispatchQueue = dispatch_queue_create("com.amazonaws.AWSSynchronizedMutableDictionary", DISPATCH_QUEUE_SERIAL);
-    }
+    self = [self initWithDictionary:@{}.mutableCopy];
+    return self;
+}
 
+- (instancetype)initWithDictionary:(NSMutableDictionary *)dictionary {
+    dispatch_queue_t queue = dispatch_queue_create("com.acme.atomic-age.dictionary", DISPATCH_QUEUE_CONCURRENT);
+    NSUUID *syncKey = [NSUUID new];
+    self = [self initWithDictionary:@{}.mutableCopy queue:queue syncKey:syncKey];
```

Comment:
Shouldn't this be like ?
`self = [self initWithDictionary:dictionary.mutableCopy queue:queue syncKey:syncKey];`

So instead of using an empty `@{}.mutableCopy` use the dictionary passed as in the `init` which is `dictionary.mutableCopy`..

Reply:
Also when you fix this, could you utilize one of the tests to use this initializer, so that this is also covered under the unit tests..